src/SignalR/server/Core/src/HubOptionsSetup`T.cs

```diff
@@ -30,6 +30,15 @@ public void Configure(HubOptions<THub> options)
             options.StreamBufferCapacity = _hubOptions.StreamBufferCapacity;
 
             options.UserHasSetValues = true;
+
+            if (_hubOptions.HubFilters != null)
+            {
+                options.HubFilters = new List<IHubFilter>(_hubOptions.HubFilters.Count);
```

Comment:
If this isn't on the critical path just do 
```suggestion
                options.HubFilters = new List<IHubFilter>(_hubOptions.HubFilters);
```

or 
```suggestion
                options.HubFilters.AddRange(_hubOptions.HubFilters);
```

Reply:
Trust in the BCL. It [has optimizations for `ICollection` in the `new List<T>(IEnumerable<T>)` ctor](https://source.dot.net/#System.Private.CoreLib/List.cs,62).

So yeah, I'd go with that constructor option.